spacelift/resource_stack.go

```diff
@@ -383,15 +407,22 @@ func stackInput(d *schema.ResourceData) structs.StackInput {
 		ret.Description = toOptionalString(description)
 	}
 
-	foundGitlab := false
+	foundSomethingElseThanGitHub := false
 	if gitlab, ok := d.Get("gitlab").([]interface{}); ok {
 		if len(gitlab) > 0 {
-			foundGitlab = true
+			foundSomethingElseThanGitHub = true
 			ret.Namespace = toOptionalString(gitlab[0].(map[string]interface{})["namespace"])
 			ret.Provider = graphql.NewString(vcsProviderGitlab)
 		}
 	}
-	if !foundGitlab {
+	if showcase, ok := d.Get("showcase").([]interface{}); ok {
+		if len(showcase) > 0 {
+			foundSomethingElseThanGitHub = true
+			ret.Namespace = toOptionalString(showcase[0].(map[string]interface{})["namespace"])
+			ret.Provider = graphql.NewString(vcsProviderShowcases)
+		}
+	}
+	if !foundSomethingElseThanGitHub {
```

Comment:
Just a thought - could you default `ret.Provider` to `GITHUB` before doing the logic for the other VCS providers? That way you could get rid of this variable, and it means when someone is adding another provider block they don't need to remember to set `foundSomethingElseThanGitHub`.

Reply:
Good point, done.